### DISCUSSION TITLE
Why does hardware encryption perform poorly in network transmission？

### DISCUSSION THREAD

**Discussion by trymine** [openssl/openssl#24604] [Q&A]
Hi,
I recently used hardware encryption to increase the encryption speed，The following is the test result。
**openssl speed -evp aes-128-cbc -elapsed**
Hardware speed test:
type                    16 bytes     64 bytes    256 bytes   1024 bytes   8192 bytes  16384 bytes
aes-128-cbc       1027.77k     3766.24k    13816.69k    44700.15k   112807.38k   113234.67k

Software speed test:
type                 16 bytes        64 bytes      256 bytes    1024 bytes   8192 bytes  16384 bytes
aes-128-cbc     15790.38k    18420.05k    19309.31k    19540.99k   19603.46k   19589.80k
Obviously, it can be seen from the above results that the larger the block size, the more obvious the advantage of hardware acceleration.
However, when I use a secure network for data transmission, the speed of the hardware and software is not much different。
My specific method is to specify the encryption algorithm as aes-128-cbc and TLS1_2 protocol. The client continues to send 100kb data, and the server continues to receive it. Use standard functions, such as SSL_read(), SSL_write().

My doubt is whether hardware encryption is not dominant in the encryption and decryption of network transmission?
If yes, is it caused by too small block size? How to determine the current block size in network transmission encryption and decryption applications?

**Comment by trymine** [comment]
Use openssl speed for speed test, you can easily specify the size blocks, and get the encryption and decryption speed comparison results of different size blocks.
Then in the network transport layer (using SSL_read or SSL_write), there must be specified size blocks when the encryption and decryption is finally called.
I want to increase the size of this size block of the network transport layer so that my hardware acceleration engine can take advantage of its speed.
**What I want to ask is, how to specify the size blocks or input buffer size of the network transport layer using the SSL_read or SSL_write function?**

**Comment by alorbach** [comment]
Did you find a solution to this? 
I am also looking for ways to amplify the performance of SSL_Write and SSL_Read but it appears to be very difficult
